Functionality seemed without problems while testing.
Not to be fully trusted since I really struggle with writting tests. 
There was incident on Error exit with my own /exit argument.